[WIP]docs(aio): add Authors Doc Style Guide

shefalisinha opened the pull request:
**Please check if the PR fulfills these requirements**
- [ ] The commit message follows our guidelines: https://github.com/angular/angular/blob/master/CONTRIBUTING.md#commit
- [ ] Tests for the changes have been added (for bug fixes / features)
- [ ] Docs have been added / updated (for bug fixes / features)


**What kind of change does this PR introduce?** (check one with "x")
```
[ ] Bugfix
[ ] Feature
[ ] Code style update (formatting, local variables)
[ ] Refactoring (no functional changes, no api changes)
[ ] Build related changes
[ ] CI related changes
[ ] Other... Please describe:
```

**What is the current behavior?** (You can also link to an open issue here)



**What is the new behavior?**



**Does this PR introduce a breaking change?** (check one with "x")
```
[ ] Yes
[ ] No
```

If this PR contains a breaking change, please describe the impact and migration path for existing applications: ...


**Other information**:

shefalisinha committed:
docs(aio): add Authors Doc Style Guide

shefalisinha committed:
docs(aio): add Authors Doc Style Guide

Foxandxss commented:
So far so good. There are some corrections needed to be done, but I don't want to step in your toes.

shefalisinha committed:
docs(aio): add Authors Doc Style Guide

shefalisinha committed:
docs(aio): add Authors Doc Style Guide

shefalisinha committed:
docs(aio): change code examples, doc regions, code tabs, code panes, attributes for all, cleanup

shefalisinha committed:
docs(aio): change code examples, doc regions, code tabs, code panes, attributes for all, cleanup

heathkit committed:
fix(upgrade): call setInterval outside the Angular zone

This wraps the $interval service when using upgrade to run the
$interval() call outside the Angular zone. However, the callback is
invoked within the Angular zone, so changes still propagate to
downgraded components.

chuckjaz committed:
docs(aio): document the non-null assertion operator

mhevery committed:
refactor(platform-browser): Remove setGlobalVar from DOM adapter

sjtrimble committed:
fix(aio): topnav toolbar position styles

- Fixed topnav on all mobile
- Fixed topnav on all docs pages
- Absolute topnav on all marketing pages
- Cleanup and code consolidation for all top-menu styles
- Add styling to topnav links on focus

alexeagle committed:
build: Introduce Bazel build rules

So far this just compiles the core and common packages.

matsko committed:
ci: update pullapprove to reflect animation ownership

alexeagle committed:
ci: switch CircleCI to 2.0

    This lets us use a docker container to provide things like the Bazel build tool

rafaelss95 committed:
docs: fix typo in 'Routes' docs

Foxandxss committed:
build(aio): ignore .ngsummary.ts files in examples

shefalisinha committed:
docs(aio): Live examples, anchor tags, no-TOC, ```html, code example-tab-pane attributes, cleanup

shefalisinha committed:
docs(aio): Live examples, anchor tags, no-TOC, ```html, code example-tab-pane attributes, cleanup

alexeagle committed:
feat(compiler-cli): introduce synchronous codegen API

sjtrimble committed:
fix(aio): fix search box overlap for small devices (#17075)

- Adjust search box height
- Adjust search box standard width and width for smaller devices
- Fix search jump outside of specified max width

IgorMinar committed:
build(aio): upgrade ngo and purify (#17210)

petebacondarwin committed:
fix(aio): ensure all views can indicate the active node (#17194)

When more than one node matches a url, the last
node defined in the navigation.json file won. This
meant that, for instance, items in both the
TopBarNarrow and the Footer views would not
indicate that they were active.

Now, each url is associated with a map of current
nodes keyed off their view.

Closes #17022

petebacondarwin committed:
refactor(aio): use the SelectMenuComponent for all select menus

The API filters and the docs version switcher now use
the SelectMenuComponent.

Fixes #16367 and #17055

Foxandxss committed:
build(aio): update features and presskit marketing pages

Foxandxss committed:
build(aio): migrate contribute page to html

Foxandxss committed:
docs(aio): remove tutorial from docs landing page

gkalpak committed:
fix(aio): fix and clean up topbar styling

Restore the changes introduced in #17075, which wre accidentally overwritten
while rebasing #17155. Also, simplify the topbar positioning rules.

gkalpak committed:
fix(aio): make `ScrollSpy` respond quicker to `scroll` events

Fixes ##17220

sjtrimble committed:
fix(aio): fix button alignment and jump

- Fix alignment on get started button on home page
- Remove jump by removing arrow from learn more button on home page

gkalpak committed:
fix(aio): fix scrolling to an element

Previously, the top-bar's height wasn't taken into account when scrolling an
element into view. As a result, the element would be hidden behind the top-bar.
Taking the top-bar height into account was not necessary before #17155, because
the top-bar was not fixed (i.e. it scrolled away).

This commit fixes the scrolling behavior by accounting for the top-bar's height
when scrolling an element into view.

(This partially reverts #17102.)

Fixes #17219
Fixes #17226

matsko committed:
perf(animations): do not create a closure each time a node is removed

gkalpak committed:
fix(aio): adjust SideNav top to avoid extra whitespace

Partially addresses #17096.

wardbell committed:
fix(aio): re-color code pretty print

closes #17246

petebacondarwin committed:
fix(aio): remove `...` separator from search results

An ellipsis was used to separate the most relevant search
results from the alphabetic list. The separator was confusing
because it was not clear what it represented.

This has been removed and the most relevant results are now
indicated by styling with a more bold font and a bit of whitespace
between them and the rest of the results.

To keep things consistent, if there are fewer than 5 results all the
results are now displayed as priorityPages.

Closes #17233

petebacondarwin committed:
docs(aio): use h1 rather than divs and hide heading anchors

In the marketing pages we do not want to show heading anchors on hover.
Previously, this was achieved by using div rather than heading elements.
Now we can use semantically accurate headings while hiding the anchor.

Closes #17244
Closes #17264

gkalpak committed:
fix(aio): expand the main content width when there is no ToC

Previously, the main content would always leave a 18% margin on the right to be
occupied by the ToC (even if there was no ToC).
This commit lets the main content expand to the right to occupy all the
available space when there is no ToC.

Fixes #17205
Fixes #17270

petebacondarwin committed:
feat(aio): do not display anchor if heading has `no-anchor` class

sjtrimble committed:
fix(aio): update contrast on mediumgray color

petebacondarwin committed:
feat(aio): add `aio-select` component

Provide the functionality for select menus in a single reusable component.

wardbell committed:
docs: security guide images
closes #17263

wardbell committed:
docs(aio): add “nougat” to compat chart

closes #17262

wardbell committed:
docs: "TypeScript to JavaScript" code tab renamed “ES6 + Decorators”

closes #17260

petebacondarwin committed:
build(aio): ignore example files that are gitignored

Rather than hard coding excludes into the dgeni config,
use the fact that we are already ignoring the boilerplate
and generated files via the .gitignore file.

matsko committed:
fix(animations): evaluate substitutions on option param values

kara committed:
fix(forms): fix min and max validator behavior on non-numbers

petebacondarwin committed:
docs(aio): do not ignore i18n example js file

petebacondarwin committed:
docs(aio): move example specific styles into router example app.css file

petebacondarwin committed:
docs(aio): remove example specific styles from the shared boilerplate styles.css

gkalpak committed:
fix(aio): smoother transition from page with SideNav to homepage (#17312)

When navigating from a page with open SideNav to a page without closed SideNav,
the main content area animates from a non-zero left margin to zero left margin.
Additionally, the top-bar on the homepage is transparent, which allows the white
background behind the main content to be seen while the left margin is animated
to zero, making it appear as if something (e.g. the SideNav covers the top-bar).

This commit works around this issue, by not making the top-bar transparent
immediately when navigating to the homepage, but animating it from its blue
color to transparent with a delay.

Fixes #17248

gkalpak committed:
fix(aio): make the ToC left hand-side bar shorter (#17280)

Make the ToC left hand-side bar shorter, so that it starts at the initial bullet
and ends at the final bullet.

Fixes #17221

petebacondarwin committed:
docs(aio): remove cli-quickstart styles.css from git

gkalpak committed:
ci: update Chromium to 464641 (Chrome v59)

gkalpak committed:
fix(aio): restrain scrolling inside ToC (when cursor over ToC)

Previously, when scrolling the ToC and reaching the top/bottom, further
mousewheel events would result in scrolling the window (and thus the main
content). This is standard browser behavior. In the case of the ToC though, the
`ScrollSpy` would detect scrolling in the main content and scroll the active ToC
to entry into view, thus resetting the scroll position of the ToC.

Reproduction:
1. Open  `~/guide/template-syntax`.
2. Start scrolling through the long ToC.
3. Try to go to the bottom of the ToC.
4. Once you reach the bottom, the main content starts scrolling down.
5. The first section ("HTML in templates") becomes "active", so the ToC is
   scrolled back up to make its corresponding entry visible.
6. Go back to step 2.

This commit improves the UX, by not allowing the main content to scroll when the
cursor is ovr the ToC and the user has scrolled all the way to the top/bottom of
it.

gkalpak committed:
fix(aio): animate hamburger in/out

Fixes #17215

petebacondarwin committed:
docs(aio): create styles.css snippet file for toh-pt5 example

Foxandxss committed:
build(aio): rename features header

sjtrimble committed:
fix(aio): api list column overflow fix (#17300)

wardbell committed:
fix(aio): scroll to top immediately when doc changes (#17310)

sjtrimble committed:
fix(aio): alert, subsection and callout width fix (#17303)

sjtrimble committed:
fix(aio): remove glossary l-sub-section class (#17305)

sjtrimble committed:
fix(aio): contributor img offset and hover fix (#17338)

sjtrimble committed:
fix(aio): copy code button overlap fix on mobile (#17302)

sjtrimble committed:
fix(aio): style and fix hover color for features cta button (#17342)

petebacondarwin committed:
docs(aio): remove example specific `.gitignore` files

All the files that should be ignored are defined in the
`aio/content/examples/.gitigore` folder.

alxhub committed:
release: cut the 4.2.0 release

alxhub committed:
docs: add changelog for 4.2.0

IgorMinar committed:
feat(aio): update the home page and docs landing page content (#17325)

Closes #17268
Closes #17230

wardbell committed:
fix(aio): sidebar folder state after select item

Closes #17245 and #17253

When the user selects a doc item in the side nav:
1) expand folder(s) leading to the selected doc item
2) on a wide display, keep other already expanded folders open
3) on narrow (mobile) display, collapse other expanded folders

Used to do (3) when wide. Issue #17245 asks for (2).

That logic was bypassed for selected node when we allowed headers to have content
because that unintentionally expanded the header’s folder when selected.
Because the selected node is no longer a header with content, removing this exclusion
also means that folders are expanded/collapsed with above logic even for API pages.

shefalisinha committed:
docs(aio): sub-section, format nocode not in use, change to prose

shefalisinha committed:
Merge branch 'aio-shefali-docs-style-guide' of https://github.com/ideablade/angular into aio-shefali-docs-style-guide

Foxandxss commented:
Ahhhh messed up. My bad. Let's start over.
